### PR TITLE
Change the parser grammar of the SelectorParser.jj file

### PR DESCRIPTION
**Pull request by AroshaMalithi**
Change the Parser grammar of the message broker.New parser grammar help to create message selectors with Comparison Operators.These java classes are support for new parser file.

### CODE REVIEW

**Comment by ThishaniLucas** [comment]
Please provide appropriate title and commit messages.

File: modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/selector/NOTEQUAL.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        Object leftValue = left.evaluate(metadata);
+        Object rightValue = right.evaluate(metadata);
+        if (leftValue == null || rightValue == null) {
+            return false;
+        }
+
+        if (rightValue != leftValue) {
+            return true;
+        }
+
```
Comment (ThishaniLucas) [inline modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/selector/NOTEQUAL.java]: Please remove unnecessary empty lines. Check that in other places as well.

File: modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/selector/NOTEQUAL.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+package io.ballerina.messaging.broker.core.selector;
+
+
+import io.ballerina.messaging.broker.core.Metadata;
+
+/**
+ * Implementation of a boolean expression. Here we compare two expressions and evaluate to a boolean value.
+ */
+public class NOTEQUAL implements BooleanExpression {
```
Comment (ThishaniLucas) [inline modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/selector/NOTEQUAL.java]: Please rename the class with something like NotEqual. All-caps is usually used for naming constants.

Comment (abeykoon) [inline modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/selector/NOTEQUAL.java]: Does not comply with class naming conventions

File: modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/selector/NOTEQUAL.java (hunk 1 above)
Comment (AroshaMalithi) [reply modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/selector/NOTEQUAL.java]: Ok thank you.I'll correct those mistakes. @ThishaniLucas.

File: modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/selector/GreaterOrEqual.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package io.ballerina.messaging.broker.core.selector;
```
Comment (madushadhanushka) [inline modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/selector/GreaterOrEqual.java]: Need to add license header

File: modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/selector/LessOrEqual.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package io.ballerina.messaging.broker.core.selector;
```
Comment (madushadhanushka) [inline modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/selector/LessOrEqual.java]: Add license header here as well

**Comment by madushadhanushka** [comment]
Please remove or reformat the PR default template on the PR description.

File: modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/selector/GreaterOrEqual.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    @Override
+    public boolean evaluate(Metadata metadata) {
+        Object leftValue = left.evaluate(metadata);
+        Object rightValue = right.evaluate(metadata);
+        if (leftValue == null || rightValue == null) {
+            return false;
+        }
+        String s = String.valueOf(leftValue);
+        String s1 = String.valueOf(rightValue);
+        double x = s.compareTo(s1);
```
Comment (madushadhanushka) [inline modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/selector/GreaterOrEqual.java]: Here line 22 to 28 repeated in four other places. Is there any good way to implement a common function and use it?

**Pull request closed without merging**